packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -44,7 +45,20 @@ const LinkControlSearchInput = ( {
 	}
 
 	return (
-		<form onSubmit={ selectSuggestionOrCurrentInputValue }>
+		// eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+		<form
+			onSubmit={ selectSuggestionOrCurrentInputValue }
+			onKeyDown={ ( event ) => {
+				event.stopPropagation();
+			} }
+			onKeyPress={ ( event ) => {
+				const { keycode } = event;
+				if ( keycode === ENTER ) {
+					selectSuggestionOrCurrentInputValue();
```

Comment:
This caught my eye. I'm sure this code got removed in earlier PRs on `LinkControl`. Does the action not get handled by the `onSubmit` handler?

If not then we could consider adding a comment here to indicate why we're having to double up here as it wasn't immediately apparent to me.

Reply:
There is this comment in the URLInput component which made this neccessary:
```
// If the suggestions are not shown or loading, we shouldn't handle the arrow keys
// We shouldn't preventDefault to allow block arrow keys navigation
```
As in this context we explicitly disable suggestions then we need to also handle stopping key propagation to observe typing. Otherwise The pop up closes.

Would you think there is a better way around this?

Reply:
I would have to also take care of the Escape button as now tests are failing b/c the link editor is not being dismissed on ESC anymore :( ugh.